app/src/sharedTestUtils/java/ch/epfl/favo/FakeViewModel.java

```diff
@@ -55,6 +56,18 @@ public CompletableFuture updateFavor(Favor favor) {
     return result;
   }
 
+  @Override
+  public void uploadOrUpdatePicture(Favor favor, Bitmap picture) {}
+
+  @Override
+  public CompletableFuture<Bitmap> downloadPicture(Favor favor) {

```

Comment:
For testing purposes you can create a setter method that takes in any bitmap and stubs it in the completablefuture result.

Then when you call downloadPicture it will actually return a bitmap object. (see AddFavorTests)

Reply:
I tried doing something like this but after some research online, if Android objects aren't in the `androidTests` directly, they are automatically set to `null`.

Reply:
You mean the bitmap object ? If so, the bitmap object can be created using Bitmap bm = Bitmap.createBitmap(200, 100, Bitmap.Config.RGB_565); for example

You can then call fakeViewModel.setImageResult(bm);
then inside setImageResult you can have something like 
CompletableFuture<Bitmap> pictureFuture = new CompletableFuture(){{complete(bm)};};

Then for the downloadPicture() method in the FakeViewModel
you can return the pictureFuture, which will behave as if you had downloaded something from Firebase